Gems/Compression/Code/Include/Compression/DecompressionInterfaceAPI.h

```diff
@@ -70,28 +70,45 @@ namespace Compression
+
+
+        //! Registers decompression interface, but does not take ownership of it
         //! If a decompression interface with a CompressionAlgorithmId is registered that
         //! matches the input decompression interface, then registration does not occur
-        //! and the unique_ptr refererence is unmodified and can be re-used by the caller
         //!
-        //! @param decompressionInterface unique pointer to decompression interface to register
-        //! @return true if the IDecompressionInterface was successfully registered, otherwise false
-        virtual bool RegisterDecompressionInterface(AZStd::unique_ptr<IDecompressionInterface>&&) = 0;
+        //! Registers decompression interface, but does not take ownership of it
+        //! @param algorithmId Unique id to associate with decompression interface
+        //! @param decompressionInterface decompression interface to register
+        //! @return true if the ICompressionInterface was successfully registered
+        virtual bool RegisterDecompressionInterface(CompressionAlgorithmId algorithmId, IDecompressionInterface&) = 0;
 
-        //! Unregisters the Decompression Interface with the specified Id if it is registered with the factory
+        //! Unregisters the decompression interface with the specified id
         //!
-        //! @param compressionAlgorithmId unique Id that identifies the Decompression Interface
-        //! @return true if the IDecompressionInterface was unregistered, otherwise false
-        virtual bool UnregisterDecompressionInterface(CompressionAlgorithmId) = 0;
-
-        //! Queries the Decompression Interface with the compression algorithmd Id
-        //! @param compressionAlgorithmId unique Id of Decompression Interface to query
-        //! @return pointer to the DeCompression Interface or nullptr if not found
-        virtual IDecompressionInterface* FindDecompressionInterface(CompressionAlgorithmId) const = 0;
+        //! @param decompressionAlgorithmId unique Id that identifies the decompression interface
+        //! @return true if the unregistration is successful
+        virtual bool UnregisterDecompressionInterface(CompressionAlgorithmId algorithmId) = 0;
+
+        //! Queries the decompression interface with the decompression algorithmd Id
+        //! @param decompressionAlgorithmId unique Id of decompression interface to query
+        //! @return pointer to the decompression tnterface or nullptr if not found
+        [[nodiscard]] virtual IDecompressionInterface* FindDecompressionInterface(CompressionAlgorithmId) const = 0;
+
+
+        //! Return true if there is an decompression interface registered with the specified id
+        //! @param algorithmId CompressionAlgorithmId to determine if an decompression interface is registered
+        //! @return bool indicating if there is an decompression interface with the id registered
+        [[nodiscard]] virtual bool IsRegistered(CompressionAlgorithmId) const = 0;
```

Comment:
Should match doxygen entry for ```algorithmId```
```suggestion
        [[nodiscard]] virtual bool IsRegistered(CompressionAlgorithmId algorithmId) const = 0;
```